### ISSUE TITLE
Issue with caches in cuda backend 

### ISSUE THREAD

**Issue by lukasm91** [GridTools/gridtools#1483]
Probably I am just wrong - we discussed caches many times, but this is unexpected to me, because I would think that the fill cache will just fill `p` if the bounds provide values, and same for flush, but it shouldn't reduce the compute domain. 

I use the release_v1.1 branch.

```cuda
#include <gridtools/stencil_composition/make_computation.hpp>
#include <gridtools/stencil_composition/stencil_composition.hpp>

#include <gridtools/storage/storage_facility.hpp>

using namespace gridtools;

struct functor1 {
  using p = gridtools::inout_accessor<0, extent<0, 0, 0, 0, 0, 3>>;
  using param_list = make_param_list<p>;
  template <typename Evaluation>
  GT_FUNCTION static void apply(Evaluation eval) {
    if (threadIdx.x == 0 && threadIdx.y == 0 && blockIdx.x == 0 &&
        blockIdx.y == 0)
      printf("functor1\n");
  }
};

int main() {
  uint_t Ni = 32;
  uint_t Nj = 32;
  uint_t Nk = 15;

  using storage_info_t = storage_traits<backend::cuda>::storage_info_t<0, 3>;
  using data_store_t =
      storage_traits<backend::cuda>::data_store_t<int, storage_info_t>;

  storage_info_t info(Ni, Nj, Nk);
  data_store_t p1(info, -1, "phi");

  using arg1 = arg<0, data_store_t>;

  auto my_grid = make_grid(Ni, Nj, Nk);

  auto comp = make_computation<backend::cuda>(
      my_grid, arg1() = p1,
      make_multistage(
          execute::forward(),
          define_caches(
              cache<cache_type::k, cache_io_policy::fill_and_flush>(arg1())),
          make_stage<functor1>(arg1())));
  comp.run();
}
```

# Actual Output

Functor is run four times:
```
> nvcc -DBOOST_ALL_NO_LIB -DBOOST_PP_VARIADICS=1 -DGT_USE_GPU -isystem=./gridtools/include -O3 -DNDEBUG   -arch=sm_70 --expt-relaxed-constexpr -std=c++14  -o out test.cu
functor1
functor1
functor1
functor1
```

# Expected output

Functor is run 15 times:

```
> ./out
functor1
functor1
functor1
functor1
functor1
functor1
functor1
functor1
functor1
functor1
functor1
functor1
functor1
functor1
functor1
```

**Comment by lukasm91** [comment]
The problem is related to caches *and* the extent.

**Comment by havogt** [comment]
resolved in #1530
